src/main/java/pro/belbix/ethparser/web3/harvest/HardWorkCalculator.java

```diff
@@ -0,0 +1,119 @@
+package pro.belbix.ethparser.web3.harvest;
+
+import org.springframework.stereotype.Service;
+import pro.belbix.ethparser.dto.HarvestDTO;
+import pro.belbix.ethparser.repositories.HardWorkRepository;
+import pro.belbix.ethparser.repositories.HarvestRepository;
+import pro.belbix.ethparser.web3.EthBlockService;
+import pro.belbix.ethparser.web3.prices.PriceProvider;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+public class HardWorkCalculator {
+
+    private final double ETH_ESTIMATE = 0.1;
+    private final PriceProvider priceProvider;
+    private final HarvestRepository harvestRepository;
+    private final HardWorkRepository hardworkRepository;
+    private final EthBlockService ethBlockService;
+
+    public HardWorkCalculator(PriceProvider priceProvider, HarvestRepository harvestRepository, HardWorkRepository hardWorkRepository, EthBlockService ethBlockService) {
+        this.priceProvider = priceProvider;
+        this.harvestRepository = harvestRepository;
+        this.hardworkRepository = hardWorkRepository;
+        this.ethBlockService = ethBlockService;
+    }
+
+    public double calculateTotalHardWorksFeeByOwner(String ownerAddress) {
+        long lastBlock = ethBlockService.getLastBlock();
+        long lastBlockDate = ethBlockService.getTimestampSecForBlock(null, lastBlock);
+        
+        List<HarvestDTO> harvests = harvestRepository.fetchAllByOwner(ownerAddress, 0, lastBlockDate);
```

Comment:
better to create another sql query with revert ordering

Reply:
Actually, in-memory sorting is not necessary. The current SQL query in `asc` order achieves the same thing that the in-memory sorting does. So I just removed the unnecessary sorting.